TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot

```diff
@@ -4,44 +4,59 @@ Resource     TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot
-    # one resource > one profile
-    Given Create A Device Profile with deviceResource
-    When Add A New resource on device profile
+ProfileResourcePOST001 - Add multiple Resources on device profile
+    # multiple resources > one profile
+    Given Set Test Variable  ${test_profile}  Test-Profile-1
+    And Generate a device profile sample  ${test_profile}
+    And Create Device Profile ${deviceProfile}
+    And Add multiple Resources on device profile
+    When Create New resource ${resourceProfile}
     Then Should Return Status Code "207"
     And Should Return Content-Type "application/json"
     And Item Index All Should Contain Status Code "201"
     And Response Time Should Be Less Than "${default_response_time_threshold}"ms
-    And New resources Should Be Added
-    [Teardown]  Delete Device Profile By Name
+    And New Resource SwitchButton in Test-Profile-1 Should Be Added
+    [Teardown]  Delete Device Profile By Name  ${test_profile}
 
-ProfileResourcePOST002 - Add multiple deviceResources on one device profile basicinfo only
-    # multiple resources > one basicinfo profile
-    Given Create A Device Profile Basicinfo Only
-    When Add multiple New resources on Device Profile Basicinfo Only
+ProfileResourcePOST002 - Add multiple Resources on multiple device profiles
+    # multiple resources > multiple profiles
+    Given Generate Multiple Device Profiles Sample
+    And Create Device Profile ${deviceProfile}
+    And Add multiple Resources on multiple device profile
+    When Create New resource ${resourceProfile}
     Then Should Return Status Code "207"
     And Should Return Content-Type "application/json"
     And Item Index All Should Contain Status Code "201"
     And Response Time Should Be Less Than "${default_response_time_threshold}"ms
-    And New resources Should Be Added
-    [Teardown]  Delete Device Profile By Name
-
-ProfileResourcePOST003 - Add multiple deviceResources on multiple device profiles
-    # multiple resources on profile basicinfo > multiple basicinfo profiles & profiles sample
-    Given Create multiple Device Profiles Basicinfo Only
-    And Create Multiple Device Profiles Sample
-    When Add multiple New resources on Device Profiles Basicinfo Only
-    Then Should Return Status Code "207"
-    And Should Return Content-Type "application/json"
-    And Item Index All Should Contain Status Code "201"
-    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
-    And New resources Should Be Added
-    [Teardown]  Delete Multiple Device Profiles By Names
+    And New Resource SwitchButton in Test-Profile-1 Should Be Added
+    And New Resource Temperature in Test-Profile-2 Should Be Added
+    And New Resource BinaryInput1 in Test-Profile-3 Should Be Added
+    [Teardown]  Delete Multiple Device Profiles By Names  Test-Profile-1  Test-Profile-2  Test-Profile-3
 
+*** Keywords ***
+New Resource ${resource_name} in ${profile_name} Should Be Added
```

Comment:
Only check the new resource name exists in the resource list.

Reply:
updated